From 3a05675661fa0e35dde0676b182b421d40f5af49 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:09:04 -0400
Subject: [powerpc] pseries: Don't try to register pseries cpu hotplug on non-pseries

Message-id: <1410545655-205645-316-git-send-email-dzickus@redhat.com>
Patchwork-id: 94633
O-Subject: [RHEL7 PATCH 315/626] powerpc/pseries: Don't try to register pseries cpu hotplug on non-pseries
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit d2a36071ef8dd24dceb95c3d9b05aaeac987b447
Author: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Date:   Tue Dec 10 11:31:02 2013 +1100

    powerpc/pseries: Don't try to register pseries cpu hotplug on non-pseries

    This results in oddball messages at boot on other platforms telling us
    that CPU hotplug isn't supported even when it is.

    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/pseries/hotplug-cpu.c b/arch/powerpc/platforms/pseries/hotplug-cpu.c
index 0ea99e3..9b8e050 100644
--- a/arch/powerpc/platforms/pseries/hotplug-cpu.c
+++ b/arch/powerpc/platforms/pseries/hotplug-cpu.c
@@ -420,4 +420,4 @@ static int __init pseries_cpu_hotplug_init(void)
 
  return 0;
 }
-arch_initcall(pseries_cpu_hotplug_init);
+machine_arch_initcall(pseries, pseries_cpu_hotplug_init);
-- 
1.7.1